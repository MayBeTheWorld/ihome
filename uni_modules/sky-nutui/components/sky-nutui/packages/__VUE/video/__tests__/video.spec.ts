import { mount } from '@vue/test-utils';
import Video from '../index.vue';

describe('Video', () => {
  test('video base info', () => {
    const wrapper = mount(Video, {
      props: {
        source: {
          src: 'xxx.mp4',
          type: 'video/mp4',
          poster:
            'https://img12.360buyimg.com/ling/s345x208_jfs/t1/168105/33/8417/54825/603df06dEfcddc4cb/21f9f5d0a1b3dad4.jpg.webp'
        },
        options: {
          controls: true,
          autoplay: true,
          muted: true,
          playsinline: true,
          loop: true
        }
      }
    });
    expect(wrapper.find<HTMLElement>('.nut-video source').html()).toContain('xxx.mp4');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
