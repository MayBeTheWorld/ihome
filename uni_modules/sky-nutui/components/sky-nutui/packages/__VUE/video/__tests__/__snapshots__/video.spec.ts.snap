// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video video base info 1`] = `
"<div class=\\"nut-video\\"><video class=\\"nut-video-player\\" autoplay=\\"\\" loop=\\"\\" controls=\\"\\" playsinline=\\"true\\" webkit-playsinline=\\"true\\" x5-video-player-type=\\"h5-page\\" x5-video-player-fullscreen=\\"false\\">
    <source src=\\"xxx.mp4\\" type=\\"video/mp4\\">
  </video>
  <!--v-if-->
  <!--v-if-->
  <div class=\\"nut-video-controller show-control\\" style=\\"display: none;\\">
    <div class=\\"control-play-btn\\"></div>
    <div class=\\"current-time\\">00:00</div>
    <div class=\\"progress-container\\">
      <div class=\\"progress\\">
        <div class=\\"buffered\\" style=\\"width: 0%;\\"></div>
        <div class=\\"video-ball\\" style=\\"transform: translate3d(0px, -50%, 0);\\">
          <div class=\\"move-handle\\"></div>
        </div>
        <div class=\\"played\\"></div>
      </div>
    </div>
    <div class=\\"duration-time\\">00:00</div>
    <div class=\\"volume muted\\"></div>
    <div class=\\"fullscreen-icon\\"></div>
  </div><!-- 错误弹窗 -->
  <div class=\\"nut-video-error\\" style=\\"display: none;\\">
    <p class=\\"lose\\">视频加载失败</p>
    <p class=\\"retry\\">点击重试</p>
  </div>
</div>"
`;
